packages/datetime/src/dateRangeInput.tsx

```diff
@@ -739,10 +749,10 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
                 ? fromMomentToDate(selectedBound)
                 : undefined;
         });
-        // show only the start date if the dates overlap
-        // TODO: add different handling for the === case once
-        // allowSingleDayRange is implemented (#249)
-        return [startDate, (startDate >= endDate) ? null : endDate] as DateRange;
+        const selectedRange = this.props.allowSingleDayRange
+            ? [startDate, (clearTime(startDate) > clearTime(endDate)) ? null : endDate]
+            : [startDate, (clearTime(startDate) >= clearTime(endDate)) ? null : endDate];
```

Comment:
what about `DateUtils.areSameDay`? that's what `DateRangePicker` [uses to implement this prop](https://github.com/palantir/blueprint/blob/master/packages/datetime/src/dateRangePicker.tsx#L442-L443).